dynamodb/dynamo_lock_table.go

```diff
@@ -78,27 +77,26 @@ func LockTableCheckSSEncryptionIsOn(tableName string, client *dynamodb.DynamoDB)
 
 // Create a lock table in DynamoDB and wait until it is in "active" state. If the table already exists, merely wait
 // until it is in "active" state.
-func CreateLockTable(tableName string, tags map[string]string, readCapacityUnits int, writeCapacityUnits int, client *dynamodb.DynamoDB, terragruntOptions *options.TerragruntOptions) error {
+func CreateLockTable(tableName string, tags map[string]string, client *dynamodb.DynamoDB, terragruntOptions *options.TerragruntOptions) error {
 	tableCreateDeleteSemaphore.Acquire()
 	defer tableCreateDeleteSemaphore.Release()
 
 	terragruntOptions.Logger.Printf("Creating table %s in DynamoDB", tableName)
 
-	attributeDefinitions := []*dynamodb.AttributeDefinition{
-		&dynamodb.AttributeDefinition{AttributeName: aws.String(ATTR_LOCK_ID), AttributeType: aws.String(dynamodb.ScalarAttributeTypeS)},
-	}
-
-	keySchema := []*dynamodb.KeySchemaElement{
-		&dynamodb.KeySchemaElement{AttributeName: aws.String(ATTR_LOCK_ID), KeyType: aws.String(dynamodb.KeyTypeHash)},
-	}
-
 	createTableOutput, err := client.CreateTable(&dynamodb.CreateTableInput{
-		TableName:            aws.String(tableName),
-		AttributeDefinitions: attributeDefinitions,
-		KeySchema:            keySchema,
-		ProvisionedThroughput: &dynamodb.ProvisionedThroughput{
-			ReadCapacityUnits:  aws.Int64(int64(readCapacityUnits)),
-			WriteCapacityUnits: aws.Int64(int64(writeCapacityUnits)),
+		TableName:   aws.String(tableName),
+		BillingMode: aws.String(DYNAMODB_PAY_PER_REQUEST_BILLING_MODE),
+		AttributeDefinitions: []*dynamodb.AttributeDefinition{
+			{
+				AttributeName: aws.String(ATTR_LOCK_ID),
+				AttributeType: aws.String(dynamodb.ScalarAttributeTypeS),
+			},
+		},
+		KeySchema: []*dynamodb.KeySchemaElement{
+			{
+				AttributeName: aws.String(ATTR_LOCK_ID),
+				KeyType:       aws.String(dynamodb.KeyTypeHash),
+			},
```

Comment:
Style Nit: Can you revert these changes back to what it was, where we extracted these lists as vars first? It reduces cyclomatic complexity and is easier to read.

Reply:
Updated.